src/data-workspace/index.js

```diff
@@ -1,9 +1,24 @@
+import i18n from '@dhis2/d2-i18n'
+import { NoticeBox } from '@dhis2/ui'
+import PropTypes from 'prop-types'
 import React from 'react'
 import { MetadataContextProvider } from '../metadata/index.js'
 import { DataWorkspace as SimpleDataWorkspace } from './data-workspace.js'
 
-export const DataWorkspace = () => (
-    <MetadataContextProvider>
-        <SimpleDataWorkspace />
-    </MetadataContextProvider>
-)
+export const DataWorkspace = ({ selectionHasNoFormMessage }) => {
+    if (selectionHasNoFormMessage) {
+        const title = i18n.t('The current selection does not have a form')
+
+        return <NoticeBox title={title}>{selectionHasNoFormMessage}</NoticeBox>
+    }
```

Comment:
Well, currently we are loading the metadata in the workspace so that will prevent loading of that. However, we have to move that up to the app-level so the context-selector has access to that metadata once we implement the new API. So I think this is fine.